drivers/mpsl/flash_sync/flash_sync_mpsl.c

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <errno.h>
+
+#include <mpsl.h>
+#include <mpsl_timeslot.h>
+#include <hal/nrf_timer.h>
+
+#include "multithreading_lock.h"
+#include "soc_flash_nrf.h"
+
+#define LOG_LEVEL CONFIG_FLASH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(flash_sync_mpsl);
+
+/* The request length specified by the upper layers is only time required to do
+ * the flash operations itself. Therefore we need to add some additional slack
+ * to each timeslot request.
+ */
+#if defined(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)
+#define TIMESLOT_LENGTH_SLACK_US 1000
+#else
+#define TIMESLOT_LENGTH_SLACK_US 100
+#endif
+
+struct mpsl_context {
+	/* This semaphore is taken with a timeout when the flash operation starts. */
+	struct k_sem timeout_sem;
+	mpsl_timeslot_session_id_t session_id; /* Timeslot session ID. */
```

Comment:
It's not required to document structs that are not part of the API.
So don't have to do this, especially if you are not adding any information.
(No action required).